evalml/data_checks/target_leakage_data_check.py

```diff
@@ -11,25 +13,49 @@
 
 
 class TargetLeakageDataCheck(DataCheck):
-    """Check if any of the features are highly correlated with the target."""
+    """Check if any of the features are highly correlated with the target by using mutual information or Pearson correlation."""
 
-    def __init__(self, pct_corr_threshold=0.95):
-        """Check if any of the features are highly correlated with the target.
+    def __init__(self, pct_corr_threshold=0.95, pearson_corr=False):
+        """Check if any of the features are highly correlated with the target by using mutual information or Pearson correlation.
 
-        Currently only supports binary and numeric targets and features.
+        If `pearson_corr=False`, this data check uses mutual information and supports all target and feature types.
+        Otherwise, it uses Pearson correlation and only supports binary with numeric and boolean dtypes.
+        Pearson correlation returns a value in [-1, 1], while mutual information returns a value in [0, 1].
 
         Arguments:
             pct_corr_threshold (float): The correlation threshold to be considered leakage. Defaults to 0.95.
+            pearson_corr (bool): Whether or not to use the Pearson correlation versus mutual information. Defaults to False, which uses mutual information.
 
         """
         if pct_corr_threshold < 0 or pct_corr_threshold > 1:
             raise ValueError("pct_corr_threshold must be a float between 0 and 1, inclusive.")
         self.pct_corr_threshold = pct_corr_threshold
+        self.pearson = pearson_corr
+
+    def _calculate_pearson(self, X, y):
+        highly_corr_cols = []
+        X_num = X.select(include=numeric_and_boolean_ww)
+        if y.logical_type not in numeric_and_boolean_ww or len(X_num.columns) == 0:
+            return highly_corr_cols
+        X_num = _convert_woodwork_types_wrapper(X_num.to_dataframe())
+        y = _convert_woodwork_types_wrapper(y.to_series())
+        highly_corr_cols = [label for label, col in X_num.iteritems() if abs(y.corr(col)) >= self.pct_corr_threshold]
+        return highly_corr_cols
+
+    def _calculate_mutual_information(self, X, y):
+        highly_corr_cols = []
+        for col in X.columns:
+            cols_to_compare = _convert_to_woodwork_structure(pd.DataFrame({col: X[col], str(col) + "y": y}))
+            mutual_info = cols_to_compare.mutual_information()
+            if len(mutual_info) > 0 and mutual_info['mutual_info'].iloc[0] > self.pct_corr_threshold:
+                highly_corr_cols.append(col)
+        return highly_corr_cols
 
     def validate(self, X, y):
-        """Check if any of the features are highly correlated with the target.
+        """Check if any of the features are highly correlated with the target by using mutual information.
 
-        Currently only supports binary and numeric targets and features.
+        If `pearson_corr=False`, supports all target and feature types. Otherwise, only supports binary with numeric and boolean dtypes.
```

Comment:
A bit confusing since `pearson_corr` is an argument of init and doesn't mean much here. Maybe having `pearson_corr` and `pearson` be named the same will help a bit? 🤷